src/service/transfer.ts

```diff
@@ -0,0 +1,52 @@
+import { AxiosResponse } from 'axios'
+import { KeycloakInstance } from 'keycloak-js'
+import { useKeycloak } from '@react-keycloak/ssr'
+
+import { apiClient } from 'service/apiClient'
+import { endpoints } from 'service/apiEndpoints'
+import { authConfig } from 'service/restRequests'
+
+import { TransferInput, TransferResponse } from 'gql/transfer'
+
+export const useCreateTransfer = () => {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async (data: TransferInput) => {
+    return await apiClient.post<TransferInput, AxiosResponse<TransferResponse>>(
+      endpoints.transfer.createTransfer.url,
+      data,
+      authConfig(keycloak?.token),
+    )
+  }
+}
+
+export const useEditTransfer = (id: string) => {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async (data: TransferInput) => {
+    return await apiClient.put<TransferInput, AxiosResponse<TransferResponse>>(
+      endpoints.transfer.editTransfer(id).url,
+      data,
+      authConfig(keycloak?.token),
+    )
+  }
+}
+
+export const useDeleteTransfer = (id: string) => {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async () => {
+    return await apiClient.delete<TransferResponse, AxiosResponse<TransferResponse>>(
+      endpoints.transfer.removeTransfer(id).url,
+      authConfig(keycloak?.token),
+    )
+  }
+}
+
+export function useDeleteManyTransfer(idsToDelete: string[]) {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async () => {
+    return await apiClient.post<TransferResponse, AxiosResponse<TransferResponse>>(
+      endpoints.transfer.removeManyTransfer.url,
+      idsToDelete,
+      authConfig(keycloak?.token),
+    )
+  }
+}
```

Comment:
```suggestion
```

Reply:
Thank you for the detailed guidelines! I've removed delete-many functionality and components and have created custom select components - e.g. SelectTransfer, SelectSourceCampaign, SelectSourceVault... Currency is imported from a common file - 'gql/currency'  and the initial currency value is set to be Currency.BGN.

Reply:
@nradkova Awesome! Thank you